apisix/plugins/consumer-restriction.lua

```diff
@@ -67,26 +83,30 @@ function _M.check_schema(conf)
     return true
 end
 
+
 function _M.access(conf, ctx)
-    if not ctx.consumer then
+    if not conf.type then
         return 401, { message = "Missing authentication or identity verification." }
     end
 
+    local type_id = type_funcs[conf.type](ctx)
```

Comment:
`type_id` is not a good name, and the `type_id` maybe a `nil` value.

we need to check it first.

Reply:
fixed